interview-questions.md

```diff
@@ -9,59 +9,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
-
-  Researched answer:
+  Your answer: The word `this` has a meaning for an key to an object; it refers to the object that owns the method,  a property of an object. 
 
+  Researched answer: In JavaScript the word "this" is a keyword referencing an object whose function is a property; the object that is currently calling the function. It has different vaules depending on where and how the function is excuted in the code, and can have the behavior of strict and non-strict modes, can be used in global and function contexts; will alway reference a single object.  
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: React (a web platform) is JavaScript open source library usedn for building user interfaces (UI); it is used to build single page or mobile applications. 
 
+  Researched answer: React is a declarative, efficient, and flexible JavaScript library for building user interfaces, with several different kinds of components; can help manage data. 
+  
+   React is used to easily create a highly interactive application with a UI that responds quickly to constantly changing data, it allows developers to create large webe applications that can change data, without re-loading the page; it is fast scalable and simple 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: Render() The render() method is the only required method in a class component.
 
-  Researched answer:
+  Researched answer: React has hree phases: Mounting, Updating and Unmounting. Under the umbrella of mouting, the 3rd componet is the render () method, is required and will alway be callled; they other 3 componets (constructor, getDerivedStateFromProps() and componentDidMount()), are optional. 
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: Gives the ability to write and add HTML in React via Javascript, converting HTML tags into react elements. 
 
-  Researched answer:
+  Researched answer: JavaScript XML (JSX), is an extension to the JavaScript language syntax, alloiwng the useer to write HTML directly within the Javascrip code.  Similar in appearance to HTML, JSX provides a way to structure component rendering using syntax familiar to many developers. The main difference between HTML and JSX is that in JSX, you need to return a single parent element, or with will not compile, in HTML, you do not have to return a parent element. 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Yarn - downloadable intructions within a package use to generate codes for React; can be used on local workspace. 
 
+  Researched answer: Yarn is a software packaging that allows the use of other develpers solutions to different problems, making it easier software develpment. The code sharing is via a package. 
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.